apps/src/dance/lab2/utils/getInitialSources.ts

```diff
@@ -0,0 +1,43 @@
+import {START_BLOCKS} from '@cdo/apps/constants';
+import {
+  getAppOptionsEditBlocks,
+  getAppOptionsEditingExemplar,
+  getAppOptionsViewingExemplar,
+} from '@cdo/apps/lab2/projects/utils';
+import {LevelProperties, ProjectSources} from '@cdo/apps/lab2/types';
+
+const isStartMode = getAppOptionsEditBlocks() === START_BLOCKS;
+const isEditingExemplar = getAppOptionsEditingExemplar();
+const isViewingExemplar = getAppOptionsViewingExemplar();
+
+/**
+ * Computes which initial sources to present based on level and project information
+ */
+export default function <T extends ProjectSources>(
```

Comment:
This is pretty similar to and partially adapted from Codebridge's `useInitialSources` - my hope is to eventually extract this logic and source handling to a common layer so I split out this function for now (FYI @molly-moen)